probe-rs/src/architecture/arm/core/armv7a.rs

```diff
@@ -286,6 +287,10 @@ impl<'probe> CoreInterface for Armv7a<'probe> {
     }
 
     fn run(&mut self) -> Result<(), Error> {
+        if matches!(self.state.current_state, CoreStatus::Running) {
```

Comment:
This wont work properly in all cases since our unknown state is running. So if we attach in halted state this wont be able to ever run again :)

Reply:
That's not accurate - when we create the core for the first time we read the current state and update it (see the logic in `new()`.  I've verified this case is handled correctly in my testing.

Reply:
Ah sorry about that, I was going off the default impl, which can be dangerous :)